plugins/colored-man-pages/colored-man-pages.plugin.zsh

```diff
@@ -1,39 +1,51 @@
-if [[ "$OSTYPE" = solaris* ]]
-then
-	if [[ ! -x "$HOME/bin/nroff" ]]
-	then
-		mkdir -p "$HOME/bin"
-		cat > "$HOME/bin/nroff" <<EOF
-#!/bin/sh
-if [ -n "\$_NROFF_U" -a "\$1,\$2,\$3" = "-u0,-Tlp,-man" ]; then
-	shift
-	exec /usr/bin/nroff -u\$_NROFF_U "\$@"
-fi
-#-- Some other invocation of nroff
-exec /usr/bin/nroff "\$@"
-EOF
-		chmod +x "$HOME/bin/nroff"
-	fi
-fi
+# Requires colors autoload.
+# See termcap(5).
+
+# Set up once, and then reuse. This way it supports user overrides after the
+# plugin is loaded.
+typeset -AHg less_termcap
+
+# bold & blinking mode
+less_termcap[mb]="${fg_bold[red]}"
+less_termcap[md]="${fg_bold[red]}"
+less_termcap[me]="${reset_color}"
+# standout mode
+less_termcap[so]="${fg_bold[yellow]}${bg[blue]}"
+less_termcap[se]="${reset_color}"
+# underlining
+less_termcap[us]="${fg_bold[green]}"
+less_termcap[ue]="${reset_color}"
+
+# Absolute path to this file's directory.
+typeset __colored_man_pages_dir="${0:A:h}"
 
 function colored() {
-	command env \
-		LESS_TERMCAP_mb=$(printf "\e[1;31m") \
-		LESS_TERMCAP_md=$(printf "\e[1;31m") \
-		LESS_TERMCAP_me=$(printf "\e[0m") \
-		LESS_TERMCAP_se=$(printf "\e[0m") \
-		LESS_TERMCAP_so=$(printf "\e[1;44;33m") \
-		LESS_TERMCAP_ue=$(printf "\e[0m") \
-		LESS_TERMCAP_us=$(printf "\e[1;32m") \
-		PAGER="${commands[less]:-$PAGER}" \
-		_NROFF_U=1 \
-		PATH="$HOME/bin:$PATH" \
-			"$@"
+  local -a environment
+
+  # Convert associative array to plain array of NAME=VALUE items.
+  local k v
+  for k v in "${(@kv)less_termcap}"; do
+    environment+=( "LESS_TERMCAP_${k}=${v}" )
+  done
+
+  # Prefer `less` whenever available, since we specifically configured
+  # environment for it.
+  environment+=( PAGER="${commands[less]:-$PAGER}" )
+
+  # See ./nroff script.
+  if [[ "$OSTYPE" = solaris* ]]; then
+    environment+=(
+      PATH="${__colored_man_pages_dir}:$PATH"
```

Comment:
I'd prefer if we hardcode `$ZSH/plugins/colored-man-pages` here. I don't love creating a hidden global variable just for this (the path only changes if the plugin is moved to a different location).

Reply:
I won't prefer that, and the reason is very simple. As it is, the plugin is perfectly drag-and-dropable into any blank zsh configuration without bringing in the whole oh-my-zsh bloatware. Its only dependency is `autoload colors`.

AFAIK, unfortunately, there is no way to retrieve the file path of a function once it is parsed and loaded, and there are no local variables at the file level (when calling `source ...`, at least); but if you know something that I don't — I'm all ears.

Reply:
Ok

Reply:
@ratijas If `autoload colors` is a dependency, shouldn't this plugin autoload that function?

I noticed that recently colored-man-pages broke for me and I don't know why. I can fix it by putting `autoload colors; colors` in my zshrc, but I never had that in my zshrc before and it was working.

I noticed that `plugins/zsh-navigation-tools/n-help` autoloads colors too, so shouldn't this plugin do the same thing so that it works without configuration?